### DISCUSSION TITLE
Code-Server not loading on IPhone

### DISCUSSION THREAD

**Discussion by prizzledev** [coder/code-server#3303] [Q&A] [answered]
Hey,
I run Code-Server now for 2 weeks ok my raspberry pi 4. It works perfectly, until I tested it on my 2nd Phone(IPhone 7+, IOS 13.3, Jailbroken with Checkra1n) (it works all the way but only on this one device I got trouble). In Browser (Safari, Chrome) I can access the login page and enter my password. After that the page stays white and isn't loading anything. I tried the App([Serverediter for code-server](https://apps.apple.com/de/app/servediter-for-code-server/id1504491325)) and there it also don't works. Is that a jailbreak detection or something else? If we don't find a solution it isn't that bad I just wanted to know if somebody else had this problem.![image](https://user-images.githubusercontent.com/82979675/117373737-20b18b80-aecc-11eb-9ef2-0f384c4a1f78.png)![image](https://user-images.githubusercontent.com/82979675/117373742-260ed600-aecc-11eb-8c23-65cc1ec398d2.png)

**Comment by jsjoeio** [comment, marked as answer]
What version of code-server are you using? We did have an issue with a blank white screen previously. 3.9.3 should have the fix!

**Reply by prizzledev** [reply]
Im using the newest version.
`code-server --version`
`3.9.3 fe2dc2deb08e378069891b622bb62ad1d261d1b1`
This is what i get.

**Reply by jsjoeio** [reply]
Hmm... It might help to show us the browser logs. ICYDK, here's a link for seeing the [logs on an iPhone on Safari](https://www.browserstack.com/guide/how-to-debug-on-iphone).

**Reply by prizzledev** [reply]
![image](https://user-images.githubusercontent.com/82979675/117588435-8c416600-b123-11eb-8815-8d457e4731b8.jpeg)![image](https://user-images.githubusercontent.com/82979675/117588440-95323780-b123-11eb-8c9b-a6646b34f135.jpeg)This is after typing in my password and submit it. It looks like there are HTML/JavaScript elements, but screen is still white.

**Reply by jsjoeio** [reply]
Super helpful! So `navigator` is undefined which is interesting 🤔 I think the culprit is the regular expression error. I feel like this is probably due to an older version of Safari running on your iPhone. The latest version of Safari supports code-server, but I don't think we support older versions like this one.

Unfortunately, I don't think there's much we can do here since this is most likely with your Safari version and not code-server 😢

**Reply by mward-sudo** [reply]
I can confirm that the latest version of code-server works on iPhone 7 (iOS 14.4.3 - non-jailbroken).

I know it's incompatible with running a jailbreak, but when running advanced web apps it's almost always going to require a relatively up-to-date version of your web browser.

**Reply by prizzledev** [reply]
Ok, like I said it's not a big problem for me, thank you for your help. <3
